### DISCUSSION TITLE
Model Predict interpretation and Stacking

### DISCUSSION THREAD

**Discussion by elopezfune** [sebp/scikit-survival#169] [Q&A]
In the documentation is explained the role of the attribute **predict** for all the models; however, it is quite confusing since the main task of survival analysis is to estimate risks.

RandomForest.predict(X) gives the ensemble risk score equal to the total number of events, which can be estimated by the sum of the estimated ensemble cumulative hazard function 𝐻̂𝑒

ComponentwiseGradientBoosting.predict(X) gives: If loss=’coxph’, predictions can be interpreted as log hazard ratio corresponding to the linear predictor of a Cox proportional hazards model. If loss=’squared’ or loss=’ipcwls’, predictions are the time to event.

SurvivalTrees.predict(X) gives: the risk score is the total number of events, which can be estimated by the sum of the estimated cumulative hazard function 𝐻̂ℎ

SVM.predict() gives: Predict risk score of experiencing an event.

If all the models give different predictions, how are we suppose to stack them?

**Comment by sebp** [comment]
Stacking doesn't just average the predicted risk scores, it uses a separate survival model that is trained on the individual risk scores. Therefore, it automatically learns how to interpret them.
